src/scripts/list_pipelines.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A script to detect all pipelines declared in pipelines/ and cache/ and
+# dump a summary of their configurations within a `tmp/` folder in the
+# project root.
+
+import sys
+import os
+import json
+
+sys.path.append(os.getcwd())
```

Comment:
Any reason this statement is nestled in the middle of the imports?

Reply:
I don't think this is necessary, if we just say that scripts must be run from the `src` folder

Reply:
Without this Python won't know where to find the `lib` module since the Python path will contain `src/scripts` and not `src` (note this statement is also in `nuts_breakdown.py` under the same folder). So it won't execute without this even if run from the `src/` folder.

A better alternative might be to use the following:
```python
path = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
sys.path.append(path)
```

This would update the Python path relative to the script location allowing it to be run from anywhere.

Reply:
That's strange. I'm able to run the scripts when my terminal is at the `src` folder so it would be good to understand better what's really going on.

Although adding the scripts to the path resolves this problem altogether and lets you run the scripts from anywhere so I'm definitely not opposed to that. I had not added it myself because I wasn't sure if manually messing with the sys.path was a great practice.

Reply:
That is strange. Is the `src` folder in your `PYTHONPATH` environment variable? And running `python scripts/nuts_breakdown.py` works when you remove the the `sys.path.append(...)` statement on line 23?